install/kubernetes/cilium/templates/cilium-configmap.yaml

```diff
@@ -439,36 +439,34 @@ data:
 {{- else if .Values.aksbyocni.enabled }}
   routing-mode: "tunnel"
   tunnel-protocol: "vxlan"
-{{- end }}
-
-{{- if eq .Values.tunnel "disabled" }}
+{{- else if .Values.routingMode }}
+  routing-mode: {{ .Values.routingMode | quote }}
+{{- else }}
+  {{- if eq .Values.tunnel "disabled" }}
   routing-mode: "native"
-{{- else if eq .Values.tunnel "vxlan" }}
+  {{- else if eq .Values.tunnel "vxlan" }}
   routing-mode: "tunnel"
   tunnel-protocol: "vxlan"
-{{- else if eq .Values.tunnel "geneve" }}
+  {{- else if eq .Values.tunnel "geneve" }}
   routing-mode: "tunnel"
   tunnel-protocol: "geneve"
-{{- else if not (or .Values.gke.enabled .Values.aksbyocni.enabled) }}
```

Comment:
i'm a bit worried about backporting this change to v1.14 branch. it changes the routing mode between patch versions v1.14.2 and v1.14.3 as @giorio94 pointed out in https://github.com/cilium/cilium-cli/issues/2070#issuecomment-1791113289. trying to figure out how to document this change in the upgrade guide 💭

Reply:
My gut feeling is that we need to fix this logic, as with the current patch it seems that it is no longer possible to override the routing mode when `gke.enabled` is true.

Reply:
IMO, enabling tunnel mode with `gke.enabled` was never supported. The docs state pretty clearly that `gke.enabled` will run direct routing mode

Reply:
[the external workload page](https://docs.cilium.io/en/stable/network/external-workloads/#enable-support-for-external-workloads) says:

> So far this functionality is only tested with the vxlan tunneling datapath mode (default for most installations).

and the external workload test is failing at `Verify cluster DNS on VM` step in cilium-cli repo when the routing mode is set to `native` like this https://github.com/cilium/cilium-cli/actions/runs/6911454905/job/18806005510?pr=2079

kind of tempted to just set the routing mode to `tunnel` using `--datapath-mode` flag like we do in cilium/cilium repo. cc @jrajahalme i guess external workload is not supported with the native routing mode? 💭💭💭

Reply:
> kind of tempted to just set the routing mode to tunnel using --datapath-mode flag like we do in cilium/cilium repo

Makes sense to me

Reply:
🚀🙏 https://github.com/cilium/cilium-cli/pull/2079